lnbits/extension_manager.py

```diff
@@ -389,21 +400,26 @@ def download_archive(self):
 
     def extract_archive(self):
         logger.info(f"Extracting extension {self.name} ({self.installed_version}).")
-        Path("lnbits", "upgrades").mkdir(parents=True, exist_ok=True)
-        shutil.rmtree(self.ext_upgrade_dir, True)
+        Path(settings.lnbits_extensions_path, "upgrades").mkdir(
+            parents=True, exist_ok=True
+        )
+
+        tmp_dir = Path(settings.lnbits_data_folder, "unzip-temp", self.hash)
+        shutil.rmtree(tmp_dir, True)
+
         with zipfile.ZipFile(self.zip_path, "r") as zip_ref:
-            zip_ref.extractall(self.ext_upgrade_dir)
-        generated_dir_name = os.listdir(self.ext_upgrade_dir)[0]
-        os.rename(
-            Path(self.ext_upgrade_dir, generated_dir_name),
-            Path(self.ext_upgrade_dir, self.id),
+            zip_ref.extractall(tmp_dir)
+        generated_dir_name = os.listdir(tmp_dir)[0]
+        shutil.rmtree(self.ext_upgrade_dir, True)
+        shutil.copytree(
+            Path(tmp_dir, generated_dir_name),
+            Path(self.ext_upgrade_dir),
         )
+        shutil.rmtree(tmp_dir, True)
```

Comment:
Same here, all this can error.

Reply:
let it fail, let fail, let it fail ❄️ ☃️ 🏔️